development.md

```diff
@@ -0,0 +1,156 @@
+# Development
+
+## Introduction
+
+OpenRazer consists of a Linux kernel module and Python daemon + library. The Linux kernel module exposes an
+sysfs interface, the Python daemon (openrazer-daemon) exposes this on it's turn over D-Bus.
+
+If you want to add support for a new hardware device it's often required to make changes to both the kernel module
+as the Python sources. For example `io.github.openrazer.openrazer.metainfo.xml` and `99-razer.rules`. The
+fake drivers can be regenerated by running `./scripts/generate_all_fake_drivers.sh -f`.
+
+This description assumes you have cloned the openrazer repository to your home directory.
+
+## Setup your development environment
+
+These instructions should help you setup a development environment for both the kernel module and Python
+daemon.
+
+### Kernel module
+
+It is possible to do development and test your changes on your local system but you risk crashes and possible data
+loss. I really recommend to do your development on a separate virtual or physical machine.
+
+Make sure you have multiple keyboards connected when you forward the Razer Keyboard to your virtual machine
+because when forwarded it isn't available to your host.
+
+I had some success with QEmu on Linux for this, you need to use:
```

Comment:
s/QEmu/QEMU/

Reply:
good one, fixed in 2c40d4b9833caa8108c949fe27f10f7435759293